src/app/services/dimDestinyTrackerService.factory.js

```diff
@@ -25,39 +32,97 @@ function DestinyTrackerService($q,
   const _reviewSubmitter = new ReviewSubmitter($q, $http, dimPlatformService, _trackerErrorHandler, loadingTracker, _reviewDataCache);
   const _reviewReporter = new ReviewReporter($q, $http, dimPlatformService, _trackerErrorHandler, loadingTracker, _reviewDataCache, _userFilter);
 
+  const _d2reviewDataCache = new D2ReviewDataCache();
+  const _d2trackerErrorHandler = new D2TrackerErrorHandler($q, $i18next);
+  const _d2bulkFetcher = new D2BulkFetcher($q, $http, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache);
+  const _d2reviewsFetcher = new D2ReviewsFetcher($q, $http, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache, _userFilter, dimPlatformService);
+  const _d2reviewSubmitter = new D2ReviewSubmitter($q, $http, dimPlatformService, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache);
+  const _d2reviewReporter = new D2ReviewReporter($q, $http, dimPlatformService, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache, _userFilter);
+
   return {
+    isDestinyOne: function() {
+      return (dimSettingsService.destinyVersion === 1);
+    },
+
+    isDestinyTwo: function() {
+      return (dimSettingsService.destinyVersion === 2);
+    },
+
     reattachScoresFromCache: function(stores) {
-      _bulkFetcher.attachRankings(null,
-                                  stores);
+      if (this.isDestinyOne()) {
```

Comment:
You can look at `stores[0].destinyVersion`

Reply:
The tracker service may get stores or items (cases where I could address the supplied objects to determine version) or it may get reviews to report (which don't have a version tied with them (but need to know which version we're talking with because different endpoints)).

Reply:
Maybe we could add a version to the reviews?